docs/src/pages/en/styling-reference.astro

```diff
@@ -0,0 +1,138 @@
+---
+import { Markdown } from '@astropub/md';
+import MainLayout from '../../layouts/MainLayout.astro';
+import manifest from '../../../../dist/custom-elements.json';
+
+const frontmatter = {
+  title: 'Styling Reference'
+};
+
+const headings = [];
+const components = await Astro.glob('./components/*');
+
+function hasLink(tagName) {
+  return components.some(c => c.file.includes(tagName));
+}
+
+const groups = [
+  'button',
+  'range',
+  'display',
+  'indicator',
+];
+
+for (let group of groups) {
+  headings.push({ depth: 2, slug: `${group}s`, text: `${capitalizeFirstLetter(group)}s` });
+}
+
+headings.push({ depth: 2, slug: `others`, text: `Others` });
+
+let allCssProps = {};
+for (let cls of getClasses(manifest)) {
+  for (let cssProp of cls.cssProperties ?? []) {
+    if (!allCssProps[cssProp.name]) {
+      allCssProps[cssProp.name] = cssProp;
+      cssProp.groups = new Set();
+    }
+    if (cls.tagName) {
+      allCssProps[cssProp.name].groups.add(cls.tagName.split('-').pop());
+    }
+  }
+}
+
+let sortedCssProps = [...Object.keys(allCssProps)].sort();
+
+function getClasses(manifest) {
+  let classes = [];
+  for (let moduleDoc of manifest.modules) {
+    classes.push(...moduleDoc.declarations.filter(d => d.kind === 'class'));
+  }
+  return classes;
+}
+
+function capitalizeFirstLetter(str) {
+  return str[0].toUpperCase() + str.slice(1);
+}
+
+function backticksToMdnLinks(str) {
+  return str.replace(
+    /`([^`]+)`/g,
+    '<a href="https://developer.mozilla.org/en-US/docs/Web/CSS/$1" target="_blank"><code>$1</code></a>'
+  );
+}
+
+---
+<MainLayout frontmatter={frontmatter} headings={headings}>
```

Comment:
Maybe can use a short intro paragraph at the top to give a little context about these variables?

Reply:
> Maybe can use a short intro paragraph at the top to give a little context about these variables?

@dylanjha I think we should hold off on any "front matter" for this PR unless you're confident this should be a standalone page (and not e.g. appended to the styling guide).

Reply:
It is it's own page, no?

https://media-chrome-docs-git-fork-luwes-css-vars-reference-mux.vercel.app/en/styling-reference

I think it makes sense as it's own page or was there another thought?

Reply:
yes, @cjpillsbury is doing the styling guide and we're thinking to put that first and then below this reference.

but separate seems okay too, I'm fine with both

Reply:
also fine with both! @dylanjha if you think it should be a separate page, then I'd agree with adding some "front matter" overview

Reply:
I vote own page cause it's pretty long

Reply:
updated this with a little description at the top. 

I added a "Reference" sidebar section that we could add to with other things. 

I kept it simple for now but maybe this could be a sidebar section that is closed by default and it can open up with a plus sign similar to how Mux docs does it or how Astro docs do it with a tabs UI thing.

https://media-chrome-docs-git-fork-luwes-css-vars-reference-mux.vercel.app/en/reference/styling